buildkite/resource_pipeline_schedule_test.go

```diff
@@ -1,358 +1,190 @@
+				if resp.Pipeline.Name == pipelineName {
+					return fmt.Errorf("Pipeline still exists: %s", pipelineName)
+				}
+				return err
+			},
+			Steps: []resource.TestStep{
+				{
+					Config: fmt.Sprintf(`
+						resource "buildkite_pipeline" "pipeline" {
+							name = "%s"
+							repository = "https://github.com/buildkite/terraform-provider-buildkite.git"
+						}
+
+						resource "buildkite_pipeline_schedule" "pipeline" {
+							pipeline_id = buildkite_pipeline.pipeline.id
+							branch = "main"
+							cronline = "%s"
+							label = "%s"
+						}
+					`, pipelineName, cronline, label),
+					Check: resource.ComposeAggregateTestCheckFunc(
+						// Schedules need a pipeline
+						loadPipeline(pipelineName, &pipeline),
+						// Confirm the schedule exists in the buildkite API
+						loadPipelineSchedule(&schedule),
+						// Confirm the schedule has the correct values in Buildkite's system
+						func(s *terraform.State) error {
+							if pipeline.Id != schedule.Pipeline.Id {
+								return fmt.Errorf("remote pipeline schedule pipeline ID (%s) doesn't match expected value (%s)", schedule.Pipeline.Id, pipeline.Id)
+							}
+							l := *schedule.Label
+							if l != label {
+								return fmt.Errorf("remote pipeline schedule label (%s) doesn't match expected value (%s)", l, label)
+							}
+							if *schedule.Cronline != cronline {
+								return fmt.Errorf("remote pipeline schedule cronline (%s) doesn't match expected value (%s)", *schedule.Cronline, cronline)
+							}
+							return nil
+						},
+						// Confirm the pipeline has the correct values in terraform state
+						resource.TestCheckResourceAttr("buildkite_pipeline_schedule.pipeline", "label", label),
+						resource.TestCheckResourceAttr("buildkite_pipeline_schedule.pipeline", "cronline", cronline),
+						resource.TestCheckResourceAttr("buildkite_pipeline_schedule.pipeline", "branch", "main"),
+					),
+				},
+				{
+					ResourceName:      "buildkite_pipeline_schedule.pipeline",
+					ImportState:       true,
+					ImportStateId:     schedule.Id,
+					ImportStateVerify: true,
+				},
+			},
+		})
+	})
 
-	return nil
+	t.Run("pipeline schedule can be updated", func(t *testing.T) {
+		var pipeline getPipelinePipeline
+		var schedule PipelineScheduleValues
+		pipelineName := acctest.RandString(12)
+		label := acctest.RandString(12)
+
+		resource.ParallelTest(t, resource.TestCase{
+			PreCheck:                 func() { testAccPreCheck(t) },
+			ProtoV6ProviderFactories: protoV6ProviderFactories(),
+			Steps: []resource.TestStep{
+				{
+					Config: fmt.Sprintf(`
+						resource "buildkite_pipeline" "pipeline" {
+							name = "%s"
+							repository = "https://github.com/buildkite/terraform-provider-buildkite.git"
+						}
+
+						resource "buildkite_pipeline_schedule" "pipeline" {
+							pipeline_id = buildkite_pipeline.pipeline.id
+							branch = "main"
+							cronline = "%s"
+							label = "%s"
+						}
+					`, pipelineName, "0 * * * *", label),
```

Comment:
This could probably be cleaned up a little by using an anonymous function with a returned string and accepting arguments, given it's used repeatedly.

```go
config := func(name, cronline, label string) string {
		return fmt.Sprintf(`
		resource "buildkite_pipeline" "pipeline" {
			name = "%s"
			repository = "https://github.com/buildkite/terraform-provider-buildkite.git"
		}

		resource "buildkite_pipeline_schedule" "pipeline" {
			pipeline_id = buildkite_pipeline.pipeline.id
			branch = "main"
			cronline = "%s"
			label = "%s"
		}
		`,name, cronline, label)
	}

```

Then your configs could just be:

```go
Config: config(pipelineName, "0 * * * *", label
```

Reply:
Yeah nice 👍

Reply:
Oh yeah nice